### ISSUE TITLE
How to train the Instruction-quality-scoring model?

### ISSUE THREAD

**Issue by iFe1er** [IronBeliever/CaR#1]
In the paper, the author describe the process as " retained the pre-trained Sentence BERT in Comet framework and directly concatenated the various parts of the instruction pair with spaces to train the IQS model".

Here is a few questions i have: 
1. I am new to NLP, it seems Comet  is a quality scoring system specilize in machine translation, why can it be used as IQS model?
2. which pretrained weights (Sentence BERT) in Comet are used?[Unbabel/wmt22-comet-da](https://huggingface.co/Unbabel/wmt22-comet-da) or[Unbabel/wmt22-cometkiwi-da](https://huggingface.co/Unbabel/wmt22-cometkiwi-da)？   
3. what's the difference between "Comet method" vs "IQS method" in the paper， is "Comet method" is also trained on your 2,541 instructions dataset? 
4. Are there any chances you can release the training script of  IQS model? Still lot of confusion here... Like the loss function, how to train the IQS model etc..

Thanks in advance !

**Comment by IronBeliever** [comment]
Thank you for your interest！

RQ1&3: The presentation in the paper may not be clear. When introducing IQE, Comet model and IQS model are both fine-tuned on the instruction dataset, the difference being that the former is more like the Estimator model architecture mentioned in the Comet paper on the QE task.
For more details you can refer to CaR/Ranking/comet/models/ranking/instruction_ranking_metric.py and CaR/Ranking/comet/models/regression/instruction_metric.py respectively.

RQ2: Pretrained weights of Sentence BERT you can find at Huggingface: sentence-transformers/all-MiniLM-L6-v2

RQ4: The training script of IQS model is described in the ‘Training of Ranking Model’ section of README.

Hope that solves your confusion.😄

**Comment by iFe1er** [comment]
Thank you for the detailed reply! It is a very insightful job!

1. So,  i think we choose Comet not because it is made for translation, but because comet is a training framework that can easily train a scoring model. In CaR setting, CaR ranking model use instruction as source,  expert rewrite respond as positive, and original respond  as a negative samples, the CaR ranking model treat it as a ranking problem and train via contrast learning. Am i correct about that?

2. If so, How to train the  "original Comet method (baseline)", which metric should we use? According to [Comet Metrics](https://unbabel.github.io/COMET/html/models.html#comet-metrics), the left upper fig looks quiet similar to IQS. Would you kindly share the major difference between comet and IQS method? (loss function and,maybe base model?)

3. For the ranking model  (base model) part, i am still a little bit confused about these two lines in IQS ranking model config:
-encoder_model: XLM-RoBERTa
-pretrained_model: xlm-roberta-large

I wonder what's the XLM-RoBERTa model here, it is not the SentenceBERT(all-MiniLM-L6-v2) mentioned before, could you kindly explain what is the difference? For example, Is IQS based on RoBERTa or SentenceBERT? or maybe when to use RoBERTa and when to use SentenceBERT? 


4. Also in IQS ranking model, 
    train_data: 
      - data/APE_score_train.csv
    validation_data: 
      - data/APE_score_valid.csv

I don't know what are these data, are there any way how can we get these two files?

Again,  very appreciated and waiting for the reply😄

**Comment by IronBeliever** [comment]
RQ1&2: Comet is actually designed for estimating translation quality (QE), not for translation itself. We borrowed this idea for Instruction Quality Estimation (IQE) task. In the IQE task, the IQS model and Comet model correspond to the Estimator model architecture and Translation Ranking model architecture in the Comet framework, respectively. In fact, our Comet model is trained using triplet margin loss, as you mentioned, while the IQS model is trained using Mean Squared Error (MSE), as described in the paper. Refer to CaR/pic/IQS&Comet.png

RQ3: Thank you for pointing out an issue. The statement in the paper is indeed incorrect 😂. The encoder_model in the Ranking part is XLM-RoBERTa, while the encoder_model in the Clustering part is SentenceBERT. We will correct this error in future versions!

RQ4: This is actually the Expert Revision Dataset.json from the [CoachLM project](https://github.com/lunyiliu/CoachLM). We selected 2,541 instructions with substantial differences before and after revision (with the calculated edit distance greater than a small threshold), and then randomly divided them into training, validation, and test sets in an 8:1:1 ratio.
